tools/tpm2_policycommandcode.c

```diff
@@ -0,0 +1,169 @@
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+#include <ctype.h>
+
+#include <tss2/tss2_sys.h>
+
+#include "files.h"
+#include "log.h"
+#include "pcr.h"
+#include "tpm2_options.h"
+#include "tpm2_policy.h"
+#include "tpm2_session.h"
+#include "tpm2_tool.h"
+#include "tpm2_util.h"
+
+typedef struct tpm2_policycommandcode_ctx tpm2_policycommandcode_ctx;
+struct tpm2_policycommandcode_ctx {
+   const char *session_path;
+   uint32_t command_code;
+   const char *out_policy_dgst_path;
+};
+
+static tpm2_policycommandcode_ctx ctx;
+
+static bool on_option(char key, char *value) {
+
+    bool result = true;
+    switch (key) {
+    case 'S':
+        ctx.session_path = value;
+        break;
+    case 'c':
+        result = tpm2_util_string_to_uint32(value, &ctx.command_code);
+        printf("%08X\n", ctx.command_code);
+        if (!result) {
+            LOG_ERR("Could not convert command-code to number, got: \"%s\"",
+                    value);
+            return false;
+        }
+        break;
+    case 'o':
+        ctx.out_policy_dgst_path = value;
+        break;
+    }
+    return true;
+}
+
+bool is_input_option_args_valid(void) {
+
+    if (!ctx.session_path) {
+        LOG_ERR("Must specify -S session file.");
+        return false;
+    }
+
+    if (!ctx.out_policy_dgst_path) {
+        LOG_ERR("Must specify -o output policy digest file.");
+        return false;
+    }
+
+    //Minimum two policies needed to be specified for compounding
+    if (!ctx.command_code) {
+        LOG_ERR("Must specify the command code.");
+        return false;
+    }
+
+    return true;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+
+    static struct option topts[] = {
+        { "session",        required_argument,  NULL,   'S' },
+        { "command-code",   required_argument,  NULL,   'c' },
```

Comment:
command code should be an argument, and allow for number of args to be greater than or equal to 1. You can create an array of CC and run policy events no?

Reply:
@williamcroberts sorry did not get you.

Reply:
example:
```
tpm2_policycommandcode 0x55
tpm2_policycommandcode 0x76
tpm2_policycommandcode 0x23
```
Could be:
```
tpm2_policycommandcode 0x55 0x76 0x23
```

Right?

Reply:
can't think of a practical use case for sequencing it like that.

Reply:
correct.